### ISSUE TITLE
feature: install only the wrapper if no template name given after --init

### ISSUE THREAD

**Issue by pabl0rg** [cbeust/kobalt#323]
Currently, if no template name is given, kobalt does not install itself in the folder.

If a template is given and a Build.kt already exists, Kobalt reports and error and quits, to prevent overwriting the existing Build.

Advantages of proposed behavior:
- projects created with the intellij plugin do not get the wrapper installed, but do have a Build.kt
- projects already using another build system may not want boilerplate template code added

**Comment by cbeust** [comment]
I'm not following what you're proposing, can you rephrase it?

What's your proposed behavior?

**Comment by cbeust** [comment]
A slight problem with this: it can work if the user is invoking an external `kobaltw` command but if they just call `./kobaltw --init` then neither `kobaltw` nor `kobalt-wrapper.jar` will be overwritten on Windows (because they are in use).

**Comment by pabl0rg** [comment]
good point.   I was thinking that --init <dir> should set up <dir> with the wrapper and and a Build.kt (if none exists).  A separate command could be used to apply templates, kind of like maven archetypes.  That would allow people to use Kobalt with their own archetypes/templates, possibly hosted on github (even as a gist).

**Comment by cbeust** [comment]
Wait a minute... isn't that exactly what `--update` does?

Well, it doesn't create a build file but it installs `kobaltw` and `kobalt-wrapper.jar`.

**Comment by pabl0rg** [comment]
Ha! I didn't know that. I thought it just updated the wrapper that I already have installed.   You may want to poll other users, but I never imagined --update could take a `<dir>` argument or that it would install kobaltw somewhere other than in the current directory

**Comment by cbeust** [comment]
No, it only works in the current directory. Basically, the flow is:

```
$ ~/bin/kobaltw --update // install in the current directory
$ ./kobaltw --init kotlin  // create project
```

**Comment by pabl0rg** [comment]
I think it's a bit counter-intuitive that `--update` would mean install, but you may want to check with the channel.

Also, it would be cool to open the template functionality to anyone on the internet (with no need to write a plugin).  In that case `--init` may also not be the best choice.  Some people know the archetype concept.  I don't know if Gradle has anything like archetypes.  

But if you want to aim Kobalt at java-outsiders (they may start showing up once they see that Kotlin is so nice), I think `--template` could be a beginner-friendly term.

**Comment by cbeust** [comment]
Interesting idea about opening up templates.

As it turns out, I already have this functionality in Kobalt today:

https://github.com/cbeust/kobalt/blob/master/modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/api/JarTemplate.kt

I could add a new option `--initZipTemplate <file>` where the file would be a zip/jar file that simply gets decompressed in the local directory.

What do you think?

**Comment by pabl0rg** [comment]
Cool! can it take a url for the jar/zip?

**Comment by cbeust** [comment]
Definitely, especially since I already have the very convenient [Kurl.kt](https://github.com/cbeust/kobalt/blob/master/modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/maven/Kurl.kt) :)

**Comment by cbeust** [comment]
Actually, I could even reuse `--init` and add that if the name of the template is a valid file, then it's assumed to be a jar/zip file, and if it starts with `http://` then it's assumed to be a jar/zip file available through http.

**Comment by pabl0rg** [comment]
Yeah, great idea.  If Kobalt supports --init <template-url> you might get two wins:

- emphasize that Kobalt is modern tool
- @dmitry-zhuravlev might be able to use them from the IJ plugin, decoupling the plugin from template generation

**Comment by cbeust** [comment]
Thinking about this, there's not much difference between  `--init file.jar` and simply extracting the jar manually, but it doesn't hurt to have all the scenarios (including http) under one unified syntax with `--init`.

**Comment by cbeust** [comment]
Done and deployed. Updated documentation:  http://beust.com/kobalt/documentation/index.html#templates
